pkg/asset/machines/gcp/machines.go

```diff
@@ -77,6 +77,22 @@ func provider(clusterID string, platform *gcp.Platform, mpool *gcp.MachinePool,
 		return nil, err
 	}
 
+	var encryptionKey *gcpprovider.GCPEncryptionKeyReference
+
+	if mpool.OSDisk.EncryptionKey != nil {
```

Comment:
Is there a reason that OSDisk.EncryptionKey can't just be an imported type from the GCP provider?  That should allow us to a do a straight assignment here.

Reply:
Yeah. `OSDisk.EncryptionKey` is defined in the types package, which represents the installer API, so we try to keep it free of external dependencies considering other projects import the types package.